packages/signature-controller/src/SignatureController.ts

```diff
@@ -292,8 +306,31 @@ export class SignatureController extends BaseControllerV2<
         'eth_sign requires 32 byte message hash',
       );
     }
+    const messageId = await this.#messageManager.addUnapprovedMessage(
```

Comment:
This new logic in each of the three `newX` methods is very similar also. Any chance we could try abstract it into a method that we call after the validation or even include the validation in a callback? Potentially:

`newUnsignedAbstractMessage(messageManager, approvalType, validateMessage, signMessage)`

Reply:
Fixed